src/connector/src/connector_common/common.rs

```diff
@@ -680,12 +682,19 @@ impl NatsCommon {
             NatsOffset::None => DeliverPolicy::All,
         };
 
-        let consumer = stream
-            .get_or_create_consumer(&name, {
-                config.deliver_policy = deliver_policy;
-                config
-            })
-            .await?;
+        let consumer = if let Ok(consumer) = stream.get_consumer(&name).await {
+            consumer
+        } else {
+            stream
+                .get_or_create_consumer(&name, {
+                    config.deliver_policy = deliver_policy;
+                    config.durable_name = Some(durable_consumer_name);
```

Comment:
Would be better if we can add some comments mentioning `durable` is the elixir here, maybe in the following PRs.